From bd490952530b0a564455a71fe9b2ef1e0378467c Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Tue, 11 Jun 2024 17:45:14 +0100
Subject: [PATCH 22/41] qt/controller: Make fullscreen controller a popup under
 Wayland

This fixes positioning both for it and context popups. Slow fade goes
away but that is a small price to pay.

Qt creates Tool windows as xdg_toplevel windows which remove the ability
to position them and then gets confused when trying to create the
context popup as it needs to know the position of its transient window
to create further popups.
Making it a popup means it gets created as an xdg popup which can be
positioned.

We need Qt 6.5 or later before this is fixed in Qt


Upstream-Status: Inappropriate

RPI-Distro repo forks original vlc and applies patches

to enable raspiberry pi support.
---

 modules/gui/qt/components/controller.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/modules/gui/qt/components/controller.cpp
+++ b/modules/gui/qt/components/controller.cpp
@@ -820,6 +820,17 @@ FullscreenControllerWidget::FullscreenCo
 
     vout.clear();
 
+#ifdef QT5_HAS_WAYLAND
+    if( b_hasWayland )
+    {
+        // Popup is less than perfect in that it seems impossible to make it non-modal
+        // and you can't get it to fade but at least it goes where it is asked to and
+        // does less confusing things with other popups
+        setWindowFlags( Qt::Popup | Qt::FramelessWindowHint);
+        setWindowModality( Qt::NonModal );
+    }
+    else
+#endif
     setWindowFlags( Qt::Tool | Qt::FramelessWindowHint | Qt::X11BypassWindowManagerHint );
     setAttribute( Qt::WA_ShowWithoutActivating );
     setMinimumWidth( FSC_WIDTH );
